src/test/lambda/commands/copyLambdaUrl.test.ts

```diff
@@ -0,0 +1,69 @@
+/*!
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import { createStubInstance, SinonStubbedInstance, stub, SinonStub, spy, SinonSpy } from 'sinon'
+import { copyLambdaUrl, NO_URL_INFO_MSG } from '../../../lambda/commands/copyLambdaUrl'
+import { LambdaFunctionNode } from '../../../lambda/explorer/lambdaFunctionNode'
+import { DefaultLambdaClient, LambdaClient } from '../../../shared/clients/lambdaClient'
+import globals from '../../../shared/extensionGlobals'
+import { addCodiconToString } from '../../../shared/utilities/textUtilities'
+
+describe('copy lambda function URL to clipboard', async () => {
+    let client: SinonStubbedInstance<LambdaClient>
+    let node: SinonStubbedInstance<LambdaFunctionNode>
+    let clipboardFunc: SinonStub
+
+    const nodeName = 'my-node-name'
+    const lambdaFuncUrl = 'the-lambda-func-url.com'
+
+    beforeEach(async () => {
+        client = createStubInstance(DefaultLambdaClient)
+        clipboardFunc = stub()
+
+        node = createStubInstance(LambdaFunctionNode)
+        stub(node, 'name').get(() => nodeName)
+    })
+
+    it('URL exists', async () => {
+        client.getFunctionUrl.resolves(lambdaFuncUrl)
+
+        await copyLambdaUrl(node, client, clipboardFunc)
+
+        assert.ok(client.getFunctionUrl.calledOnceWith(nodeName))
+        assert.ok(clipboardFunc.calledOnceWith(lambdaFuncUrl, 'URL'))
+    })
+
+    describe("URL doesn't exist", async () => {
+        let spiedInformationMessage: SinonSpy
+        let spiedStatusBarMessage: SinonSpy
+
+        before(async () => {
+            spiedInformationMessage = spy(globals.window, 'showWarningMessage')
+            spiedStatusBarMessage = spy(globals.window, 'setStatusBarMessage')
+        })
+
+        afterEach(async () => {
+            spiedInformationMessage.resetHistory()
+            spiedStatusBarMessage.resetHistory()
+        })
+
+        const emptyValues = [undefined, '']
+        emptyValues.forEach(val => {
+            it(`URL: '${val}'`, async () => {
+                client.getFunctionUrl.resolves(val)
+
+                await copyLambdaUrl(node, client, clipboardFunc)
+
+                assert.deepStrictEqual(client.getFunctionUrl.args, [[nodeName]])
+                assert.deepStrictEqual(clipboardFunc.args, [])
+                assert.deepStrictEqual(spiedInformationMessage.args, [[NO_URL_INFO_MSG]])
+                assert.deepStrictEqual(spiedStatusBarMessage.args, [
+                    [addCodiconToString('circle-slash', NO_URL_INFO_MSG), 5000],
+                ])
+            })
+        })
+    })
```

Comment:
If we change `copyLambdaUrl` to throw, then these tests can be simplified:
```suggestion
    describe("URL doesn't exist", async () => {
        const emptyValues = [undefined, '']
        emptyValues.forEach(val => {
            it(`URL: '${val}'`, async () => {
                client.getFunctionUrl.resolves(val)
                await assert.rejects(() => copyLambdaUrl(node, client, clipboardFunc), ToolkitError)
            })
        })
    })
```
If you wanted to test the error in more detail then you could do this:
```ts
const result = copyLambdaUrl(node, client, clipboardFunc).catch(e => e)
assert.ok(result instanceof ToolkitError)
assert.strictEqual(result.code, 'NoURLFound')
// etc.
```